packages/xo-server/src/api/pif.mjs

```diff
@@ -73,12 +74,34 @@ connect.resolve = {
 // ===================================================================
 // Reconfigure IP
 
-export async function reconfigureIp({ pif, mode = 'DHCP', ip = '', netmask = '', gateway = '', dns = '' }) {
-  const xapi = this.getXapi(pif)
-  await xapi.call('PIF.reconfigure_ip', pif._xapiRef, mode, ip, netmask, gateway, dns)
-  if (pif.management) {
-    await xapi.call('host.management_reconfigure', pif._xapiRef)
-  }
+export async function reconfigureIp({ pif, mode, ip = '', netmask = '', gateway = '', dns = '', ipv6, ipv6Mode }) {
+  const task = this.tasks.create({
+    name: `reconfigure ip of: ${pif.device}`,
+    objectId: pif.uuid,
+    type: 'xo:pif:reconfigureIp',
+  })
+  await task.run(async () => {
+    const xapi = this.getXapi(pif)
+
+    if ((ipv6 !== '' && pif.ipv6?.[0] !== ipv6) || ipv6Mode !== undefined) {
```

Comment:
There should probably be an error if `ipv6` and `ipv6Mode` are not consistent with each other.